zerver/actions/message_edit.py

```diff
@@ -1244,6 +1266,71 @@ def check_time_limit_for_change_all_propagate_mode(
     )
 
 
+def check_time_limit_for_change_all_or_stream_edit_request(
+    message: Message,
+    user_profile: UserProfile,
+    stream_id: int | None,
+    topic_name: str | None,
+    propagate_mode: str,
+    edit_limit_buffer: int,
+    is_resolve_topic_request: bool,
+) -> None:
+    if (
+        stream_id is not None
+        and user_profile.realm.move_messages_between_streams_limit_seconds is not None
+        and not user_profile.is_realm_admin
+        and not user_profile.is_moderator
+    ):
+        deadline_seconds = (
+            user_profile.realm.move_messages_between_streams_limit_seconds + edit_limit_buffer
+        )
+        if (timezone_now() - message.date_sent) > timedelta(seconds=deadline_seconds):
+            raise JsonableError(_("The time limit for editing this message's channel has passed"))
+
+    if (
+        propagate_mode == "change_all"
+        and not is_resolve_topic_request
+        and not user_profile.is_realm_admin
+        and not user_profile.is_moderator
+        and (topic_name is not None or stream_id is not None)
+    ):
+        check_time_limit_for_change_all_propagate_mode(message, user_profile, topic_name, stream_id)
+
+
+def check_message_or_topic_edit_permissions(
+    user_profile: UserProfile,
+    message: Message,
+    stream_id: int | None,
+    topic_name: str | None,
+    edit_limit_buffer: int,
+) -> None:
+    if not user_profile.can_move_messages_to_another_topic():
+        raise JsonableError(_("You don't have permission to edit this message"))
+
+    # If there is a change to the topic, check that the user is allowed to
+    # edit it and that it has not been too long. If user is not admin or moderator,
+    # and the time limit for editing topics is passed, raise an error.
+    if (
+        user_profile.realm.move_messages_within_stream_limit_seconds is not None
+        and not user_profile.is_realm_admin
+        and not user_profile.is_moderator
+    ):
+        deadline_seconds = (
+            user_profile.realm.move_messages_within_stream_limit_seconds + edit_limit_buffer
+        )
+        if (timezone_now() - message.date_sent) > timedelta(seconds=deadline_seconds):
+            raise JsonableError(_("The time limit for editing this message's topic has passed."))
+
+
+def check_resolve_topic_permissions(
+    user_profile: UserProfile,
+    message: Message,
+    stream_id: int | None,
+) -> None:
+    if not user_profile.can_resolve_topic():
```

Comment:
I'm not sure this function is a useful helper; leaving it in the caller seems like it might be more readable?

Reply:
that seems reasonable... it would've made sense previously when the method had some other things as well